File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    
    static boolean connected[];
    
    public int solution(int n, int[][] computers) {
        
        int answer = 0;
        connected = new boolean[n];
        
        for(int i=0; i<n; i++){
            if(connected[i]) continue;
            connected[i] = true;
            answer++;
            connect(i, computers);
        }
         
        return answer;
    }
    
    public void connect(int now, int[][] computers){
        
        for(int i=0; i<computers.length; i++){
            if(computers[now][i] == 1 && !connected[i]){
                connected[i] = true;
                connect(i, computers);
            }
        }
        
    }

}

File: 프로그래머스/3/43162. 네트워크/네트워크.js
var connected = [];

function solution(n, computers) {
   
    var answer = 0;
    connected = Array(n).fill(false);
    
    for(let i=0; i<n; i++){
        if(connected[i]) continue;
        connected[i] = true;
        answer++;
        connect(i, computers);
    }
    
    return answer;
}

function connect(now, computer){
    
    for(let i=0; i<computer.length; i++){
        if(computer[now][i] == 1 && !connected[i]){
            connected[i] = true;
            connect(i, computer);
        }
    }
    
}
    